test_framework/terraform/aws/sles/k3s_instaces.tf

```diff
@@ -98,8 +98,13 @@ resource "null_resource" "rsync_kubeconfig_file" {
   ]
 
   provisioner "remote-exec" {
-    inline = ["until([ -f /etc/rancher/k3s/k3s.yaml ] && [ `sudo /usr/local/bin/kubectl get node -o jsonpath='{.items[*].status.conditions}'  | jq '.[] | select(.type  == \"Ready\").status' | grep -ci true` -eq $((${var.lh_aws_instance_count_controlplane} + ${var.lh_aws_instance_count_worker})) ]); do echo \"waiting for k3s cluster nodes to be running\"; sleep 2; done"]
 
+    inline = [
+      "while [ ! -f /tmp/signal ]; do echo \"waiting for packages installed\"; sleep 2; done",
+      "jq --version",
```

Comment:
I don't think we need to have this, but rather a general check for cloud init

Reply:
To this, I am suspecting the `$PATH` inside the terraform_*.sh, as I am not certain how remote-exec runs.

I would have a simple check like:
```
"command -v jq || echo $PATH",
```
And maybe we should specify bash as the interpreter:
```
    inline = [
...
    ]
    interpreter = "/bin/bash"
```
ref.: https://github.com/hashicorp/terraform/issues/5397#issuecomment-190793340

Reply:
@kaxing not sure if I get your point, but I mean we don't need to check jq, but rather make this check as a general check to see if packages all installed ready by cloud-init CLI. However, I feel we should add `set -e` in the scripts used in the user_data, so we will know the TF provisioning failed before any test execution. 

About how remote-exec runs, it's not really a point for us here.

Reply:
@innobead just thinking if this is for an one-time investigation.

We can certainly enable `set -e` even `set -x` if there are no security related output in-between TF provisioning logs.
About cloud-init I'v just notice the subcommand: [collect-logs ](https://cloudinit.readthedocs.io/en/latest/topics/cli.html?highlight=collect-logs#collect-logs) grabs quite a bit information if there is suitable place to keep the logs for each job:
```
$ cloud-init collect-logs --verbose
collected cloud-init --version
collected dpkg version
collected cloud-init version: /usr/bin/cloud-init 22.1-14-g2e17a0d6-0ubuntu1~20.04.3

collected dmesg output
collected systemd journal of current boot
collected file: /var/log/cloud-init.log
collected file: /var/log/cloud-init-output.log
collected dir /run/cloud-init
Wrote /home/ubuntu/cloud-init.tar.gz
```

Reply:
Yeah, this seems helpful, but not sure if we need this. If want, you can have another TF file provisioner to copy this tar back.